themes-default/slim/src/store/modules/shows.js

```diff
@@ -40,7 +40,27 @@ const mutations = {
 };
 
 const getters = {
-    getShowById: state => ({ id, indexer }) => state.shows.find(show => Number(show.id[indexer]) === Number(id)),
+    getShowById: state => ({ id, indexer }) => {
+        const show = state.shows.find(show => Number(show.id[indexer]) === Number(id));
+        return show;
+    },
+    /**
+     * Get a show from the show array (store.shows.shows).
+     *
+     * This is the only way of utilizing vuex and have vue detect changes in the mapped computed property.
+     * If we don't pass shows into this getter, and have it check state like `getShowbyId` then within the computed property
+     * vue will not detect any changes in `state.shows`. We would have to add another watch, for that. While the computed
+     * should be able to handle that.
+     * @param {object} _ - state by is not used
+     * @param {array} shows - array with shows
+     * @param {number} id - show's id
+     * @param {string} indexer - show's indexer
+     * @returns {object} show object or null
+     */
+    getShowByIdFromShows: _ => ({ shows, id, indexer }) => {
+        const show = shows.find(show => Number(show.id[indexer]) === Number(id));
```

Comment:
@sharkykh passing in the shows, I can have vue computed take care of the reactiveness.
I worked around the "issue" before by added the `currentShow` https://github.com/pymedusa/Medusa/pull/4486/files#diff-4327d3f74ed18de447b2fc16ffe81b5eR7 object.
But this feels better. What do you think?

Reply:
Okay, umm, you created a getter that doesn't use the store state, which seems counter-intuitive (to me).
But I don't know what reactiveness issue you are trying to work around.

My point is, the way you've used this getter in `edit-show.vue` makes it pretty much **identical** to `getShowById` defined above.
I don't see a reason to pass the "shows" state around - using `mapState` to get it in `edit-show.vue`, and then passing it back to the getter that **already has access to it** from the `state` parameter.

You can just use `getShowById` IMO.

Reply:
Ok Lets user getShowById as An example.
Just swap it in. Then at mount time state.shows.shows is empty. So we'll start with An empty array. And show Will be `null`.

How would you fix that?